### PR TITLE
Feat/pisa25 364/ckeditor sourcedialog plugin

### PR DESCRIPTION
**Pull request by shaveko**
related to https://oat-sa.atlassian.net/browse/PISA25-364

⚠️ update dependencies to release versions before merging

 - [x] https://github.com/oat-sa/tao-core-ui-fe
 - [x] https://github.com/oat-sa/tao-core-shared-libs-fe

### Description

This PR enables the [sourcedialog](https://ckeditor.com/cke4/addon/sourcedialog) plugin based on environment configuration 

To enable sourcedialog .env file in project root has to be updated with the following string:

```
FEATURE_FLAG_CKEDITOR_SOURCEDIALOG=true
```

### How to test

Test with installed version of tao
kitchen link is in JIRA ticket's environment field

### CODE REVIEW

**Review by rgomeztao — APPROVED** [review]
looks nice, pretty straightforward, i left a comment optional to review, nice work!

File: models/classes/featureFlag/Repository/FeatureFlagRepository.php [hunk 1]
```diff
@@ -92,6 +92,9 @@ public function list(): array
         if (!array_key_exists(FeatureFlagCheckerInterface::FEATURE_FLAG_LISTS_DEPENDENCY_ENABLED, $output)) {
             $output[FeatureFlagCheckerInterface::FEATURE_FLAG_LISTS_DEPENDENCY_ENABLED] = false;
         }
+        
+        $sourceDialogEnabledInEnv = filter_var($_ENV['FEATURE_FLAG_CKEDITOR_SOURCEDIALOG'] ?? false, FILTER_VALIDATE_BOOLEAN) ?? false;
+        $output[FeatureFlagCheckerInterface::FEATURE_FLAG_CKEDITOR_SOURCEDIALOG] = $sourceDialogEnabledInEnv;
```
Comment (jsconan) [inline models/classes/featureFlag/Repository/FeatureFlagRepository.php:97]: **thought:** Is it actually needed??? I'd expect this to be automated. Actually, it should: https://github.com/oat-sa/tao-core/pull/3822/files#diff-1fe008b3b60d47cd1fe865ec3202300e46008d2afabfcf4564bde407ec4e3b29R83-R87

File: models/classes/featureFlag/FeatureFlagCheckerInterface.php [hunk 2]
```diff
@@ -28,6 +28,7 @@ interface FeatureFlagCheckerInterface
     public const FEATURE_FLAG_TABULAR_IMPORT = 'FEATURE_FLAG_TABULAR_IMPORT_ENABLED';
     public const FEATURE_FLAG_LISTS_DEPENDENCY_ENABLED = 'FEATURE_FLAG_LISTS_DEPENDENCY_ENABLED';
     public const FEATURE_FLAG_STATISTIC_METADATA_IMPORT = 'FEATURE_FLAG_STATISTIC_METADATA_IMPORT';
+    public const FEATURE_FLAG_CKEDITOR_SOURCEDIALOG = 'FEATURE_FLAG_CKEDITOR_SOURCEDIALOG';
```
Comment (jsconan) [inline models/classes/featureFlag/FeatureFlagCheckerInterface.php:31]: **chore:** Please document the feature flag in the readme. See https://github.com/oat-sa/tao-core/tree/develop#environment-variables

**Review by jsconan — CHANGES_REQUESTED** [review]
It works, but I'm not sure we need to add code for loading the env var.

Regarding the behavior, I think it is not correct. The source should be inline, not in a popup.
image

File: models/classes/featureFlag/FeatureFlagCheckerInterface.php (hunk 2 above)
Comment (shaveko) [reply models/classes/featureFlag/FeatureFlagCheckerInterface.php:31]: Thanks, added the doc

File: models/classes/featureFlag/Repository/FeatureFlagRepository.php [hunk 3]
```diff
@@ -92,6 +92,9 @@ public function list(): array
         if (!array_key_exists(FeatureFlagCheckerInterface::FEATURE_FLAG_LISTS_DEPENDENCY_ENABLED, $output)) {
             $output[FeatureFlagCheckerInterface::FEATURE_FLAG_LISTS_DEPENDENCY_ENABLED] = false;
         }
+        
+        $sourceDialogEnabledInEnv = filter_var($_ENV['FEATURE_FLAG_CKEDITOR_SOURCEDIALOG'] ?? false, FILTER_VALIDATE_BOOLEAN) ?? false;
```
[the remark this answers is not in the document]
Comment (shaveko) [reply models/classes/featureFlag/Repository/FeatureFlagRepository.php]: Removed the code, it works well without it

File: models/classes/featureFlag/Repository/FeatureFlagRepository.php (hunk 1 above)
Comment (shaveko) [reply models/classes/featureFlag/Repository/FeatureFlagRepository.php]: Thanks, indeed, it works well without that code.

**Review by jsconan — APPROVED** [review]
All good!
As the ACs have been updated, the behavior is acceptable.

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code respects code style rules
- [x] New code respects best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

File: models/classes/featureFlag/FeatureFlagCheckerInterface.php (hunk 2 above)
Comment (gabrielfs7) [inline models/classes/featureFlag/FeatureFlagCheckerInterface.php:31]: Maybe we do not need to have it here as it is only used in the FE and already present in the README.md

File: views/package.json [hunk 4]
```diff
@@ -15,8 +15,8 @@
         "@babel/polyfill": "^7.4.4",
         "@oat-sa/tao-core-libs": "^0.5.3",
         "@oat-sa/tao-core-sdk": "^2.0.1",
-        "@oat-sa/tao-core-shared-libs": "1.5.2",
-        "@oat-sa/tao-core-ui": "^2.0.0",
+        "@oat-sa/tao-core-shared-libs": "git+https://github.com/oat-sa/tao-core-shared-libs-fe#feat/PISA25-364/add-sourcedialog-plugin",
```
Comment (gabrielfs7) [inline views/package.json:18]: - [ ] Just a reminder to update it before merge

**Review by gabrielfs7 — APPROVED** [review]

**Pull request merged**